src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -161,6 +161,14 @@ DQMatMulCWi::DQMatMulCWi() {
             auto matched_node_qcoeff_out = uat::_(node_to_output).at_or_at_or_at(qcvtc, reshapec, qcoeff);
             auto matched_node_muls_out = uat::_(node_to_output).at_or_at(qcvtm, qmuls);
 
+            if (!ctx.get().mm_dq_full) {
+                const auto& matm_mul_out_shape = matched_matmul->get_output_shape(0);
+                const auto& matm_mul_in_shape = matched_matmul->get_input_shape(1);
+                NPUW_ASSERT(matm_mul_out_shape.back() == matm_mul_in_shape.front());
+                NPUW_ASSERT(matched_matmul->get_transpose_b());
```

Comment:
so what if they don't match? user can't do anything with this but we could

Reply:
What do you suggest to do in this case?

Reply:
I don't know, is that a legit case at all?

Reply:
so we have `get_transpose_b()`checked in the above ìf`- so nothing to worry about